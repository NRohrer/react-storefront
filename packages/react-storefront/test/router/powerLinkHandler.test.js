import powerLinkHandler from '../../src/router/powerLinkHandler'
import Response from '../../../react-storefront-moov-xdn/src/Response'

describe('powerLinkHandler', () => {
  let response
  let request

  beforeEach(() => {
    global.env = {}
    global.fns = {
      export: () => {}
    }
    request = {
      sendResponse: jest.fn()
    }
    response = new Response(request)
  })

  describe('powerLinkHandler', () => {
    it('should set response content type to javascript', () => {
      powerLinkHandler(null, {}, response)
      expect(response.get('content-type')).toBe('application/javascript')
    })
    it('should cache', () => {
      powerLinkHandler(null, {}, response)
      expect(response.get('x-moov-cache')).toBe('true')
    })
    it('should set cache age', () => {
      powerLinkHandler(null, {}, response)
      expect(response.get('cache-control')).toBe('max-age: 86400, s-maxage: 31536000')
    })
    it('should set service worker path in body', () => {
      powerLinkHandler(
        null,
        {
          protocol: 'https:',
          hostname: 'test.com',
          port: 443
        },
        response
      )
      expect(request.sendResponse.mock.calls[0][0]).toMatchSnapshot()
    })
  })
})
