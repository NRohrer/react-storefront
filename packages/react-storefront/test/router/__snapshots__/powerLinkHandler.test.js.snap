// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`powerLinkHandler powerLinkHandler should set service worker path in body 1`] = `
Object {
  "body": "
    var links = Array.from(document.querySelectorAll('a[data-rsf-power-link]')).map(function(link) {
      var powerlink = link.getAttribute('href');
      powerlink += (powerlink.indexOf('?') === -1 ? '?' : '&') + 'powerlink'
      link.setAttribute('href', powerlink);
      return link.getAttribute('href')
    });

    var el = document.createElement('iframe');

    el.setAttribute('src', 'https://test.com/pwa/install-service-worker.html?preload=' + encodeURIComponent(JSON.stringify(links)));
    el.setAttribute('style', 'height:1px;width:1px;');
    el.setAttribute('frameborder', '0');
    
    document.body.appendChild(el);
  ",
  "htmlparsed": true,
}
`;
